src/Raven.Pal/src/posix/pager.c

```diff
@@ -454,4 +537,54 @@ int32_t rvn_map_memory(void *handle,
         *mem = NULL;
     }
     return rc;
+}
+
+
+int32_t rvn_write_file_io(
+    void* handle,
+    int32_t count,
+    struct page_to_write *buffers,
+    int32_t *detailed_error_code
+)
+{
+    struct handle *handle_ptr = handle;
+    for(int32_t i = 0; i < count; i++)
+    {
+        int64_t offset = buffers[i].page_num * VORON_PAGE_SIZE;
+        int64_t size = (int64_t)buffers[i].count_of_pages * VORON_PAGE_SIZE;
+        if(rvn_pwrite(handle_ptr->file_fd, buffers[i].ptr, size, offset) <= 0)
+        {
```

Comment:
this is `unlikely` to be a branch taken. I would invert the condition and just `continue`